docs/proposals/20191017-control-plane.md

```diff
@@ -0,0 +1,569 @@
+metadata:
+  name: my-acmemachinetemplate
+  namespace: default
+spec:
+  osImage:
+    id: objectstore-123456abcdef
+  instanceType: θ9.medium
+  iamInstanceProfile: "control-plane.cluster-api-provider-acme.sigs.k8s.io"
+  sshKeyName: my-ssh-key
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: AcmeCluster
+metadata:
+  name: my-acmecluster
+  namespace: default
+spec:
+  region: antarctica-1
+```
+
+![controlplane-init-1](images/controlplane/controlplane-init-1.png)
+![controlplane-init-2](images/controlplane/controlplane-init-2.png)
+![controlplane-init-3](images/controlplane/controlplane-init-3.png)
+![controlplane-init-4](images/controlplane/controlplane-init-4.png)
+
+##### Scale Up
+
+- Allow scale up a control plane with stacked etcd to only odd numbers, as per 
+  [etcd best practice](https://etcd.io/docs/v3.3.12/faq/#why-an-odd-number-of-cluster-members).
+- However, allow a control plane using an external etcd cluster to scale up to other numbers such as 2 or 4.
+- Scale up operations must not be done in conjunction with an upgrade operation, this should be enforced using a validation webhook.
+- Scale up operations *could* be blocked based on the Health status
+  - for stacked etcd quorum must be considered
+  - for external etcd only availability of etcd should be considered.
+  - Health status is described below.
+
+![controlplane-init-6](images/controlplane/controlplane-init-6.png)
+
+##### Scale Down
+
+- Scale down a stacked control plane
+  - For each replica being deleted:
+    - Remove the replica’s etcd member from the etcd cluster
+    - Update the kubeadm generated config map
+- Scale down a control plane with an external etcd
+  - Same as above minus etcd management
+- Scale down operations must not be done in conjunction with an upgrade operation, this should not impact manual operations for recovery. This should also be enforced using a validation webhook.
+- Scale down operations *could* be blocked based on the Health status
+  - for stacked etcd quorum must be considered
+  - for external etcd only availability of etcd should be considered.
+  - Health status is described below.
+- Scale to 0 must be rejected for the initial support, see below for deletion workflows.
+
+![controlplane-init-7](images/controlplane/controlplane-init-7.png)
+
+##### Delete of the entire KubeadmControlPlane (kubectl delete controlplane my-controlplane)
+
+- Completely removing the control plane and issuing a delete on the underlying machines.
+  - Deletion is equivalent to a scale to 1 followed by a deletion of a single replica control plane.
+- User documentation should focus on deletion of the Cluster resource rather than the KubeadmControlPlane resource.
+- Cluster deletion will need to be modified to ensure KubeadmControlPlane deletion is done in the proper order.
+
+##### Cluster upgrade (using create-swap-and-delete)
+
+- Triggered on any changes to Version, InfrastructureTemplate, or the KubeadmConfiguration used.
```

Comment:
It says above the kubeadm bits should be immutable? (although I just added a question there)